frontend/src/components/SideNav.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import "../styles/SideNav.css";
+
+/**
+ *
+ * @param {Array} tabs The array of tabs that you want to populate the side nav with.
+ *                     Each element should be formated as such:
+ *                     {
+ *                      tab_id: {number from 0 to n - 1 where n is the number of tabs}
+ *                              (Note: tabs need to be labeled sequentially)
+ *                      tab_name: {the name/title that will be displayed on the side-nav bar}
+ *                      tab_content: {jsx component that will be rendered on the right-hand side of the screen}
+ *                     }
+ */
+function SideNav({ tabs }) {
+  // This is the content that is displayed on the right-hand side of the screen
+  // based on which tab was selected in the side-nav bar.
+  const [displayed, setDisplayed] = useState(tabs[0].tab_content);
+  // This helps determine which tab in the side nav bar should be highlighted.
+  const [selected, setSelected] = useState(0);
+
+  // Changes the content displayed, and the tab that gets highlighted.
+  const changeTab = (id) => {
+    setDisplayed(tabs[id].tab_content);
+    setSelected(id);
+  };
+
+  return (
+    <div className="side_nav_layout">
+      <div className="side_nav_buttons">
+        {tabs.map((tab) => (
```

Comment:
The `tab_id` field on each tab isn't strictly necessary, you could do this instead:

```js
{tabs.map((tab, tab_id) => (
```

The extra parameter `tab_id` stores the index of the tab, so you don't have to store that value manually.

---

Also React will complain if you don't give every button a unique `key` property, so I'd just add something like `key={tab.tab_name}` right below this.